### ISSUE TITLE
DNS Pod does not start 

### ISSUE THREAD

**Issue by rahulmishra** [kelseyhightower/kubernetes-the-hard-way#285]
Hello,

I have been following this guide to create a cluster on Vagrant.I am running into the following issue and i hoping someone can point me in the right direction.

Everything has worked fine when i set up the control plane and the worker nodes. I can do 

> kubectl get nodes 

And it shows me my two nodes.

I was stumped when it came to setting up a POD Network, I have one master and two slaves. Passed --cluster-cidr=10.200.0.0/16 to the controller manager

Node 1 POD CIDR 10.200.0.0/17
Node 1 eth1 address: 172.178.205.11

Node 2 POD CIDR 10.200.128.0/17
Node 2 eth1 address: 172.178.205.12

as mentioned in https://github.com/kelseyhightower/kubernetes-the-hard-way/blob/master/docs/09-bootstrapping-kubernetes-workers.md 

Now, in the first attempt i did not set any routes on the worker nodes because the default gateway was set to 10.0.2.2

> ip route list
> default via 10.0.2.2 dev eth0  proto static  metric 100

When i install the DNS Pod, it keeps going into crash back off, i tried to investigate through the logs with

>[root@master vagrant]# kubectl logs kube-dns-6c857864fb-px2v9 kubedns -n=kube-system
Error from server: Get https://k8s-slave-2:10250/containerLogs/kube-system/kube-dns-6c857864fb-px2v9/kubedns: dial tcp 92.242.140.21:10250: getsockopt: connection refused

I am not sure how that address (92.242.140.21) came into being because i am pretty sure that address is not resolvable.

Can someone point me in the right direction? What should i look for?

Regards,
Rahul

**Comment by frezbo** [comment]
@rahulmishra you either need to create the routes manually or use another networking plugin like flannel/calico and then bring up the DNS pod

**Comment by rahulmishra** [comment]
@frezbo  At this point, i don't want to use a networking plugin because most of them require me to set up an explicit POD CIDR and i chose the POD CIDR in this guide.
Could you give me some more details on what routes i should be setting.

Here is what i tried, 
On Node 1, i said 
> ip route add 10.200.128.0/17 dev eth1
On Node 2, i said
> ip route add 10.200.0.0/17 dev eth1

Here 10.200.xxx.0/17 is the slice of the POD CIDR i explicitly allocated to the worker node.My Cluster CIDR is 10.200.0.0/16

What am i doing wrong?

**Comment by petebue** [comment]
@rahulmishra do you have 10.32.0.1 in your kubernetes.pem cert?  We were just struggeling with a similar issue to yours and found that the pods communicate with the api server on port 10.32.0.1 and if it's not in your cert it will cause the DNS to fail to start up all the way.

**Comment by rahulmishra** [comment]
@petebue  Thanks for the pointer. Let me check

**Comment by rahulmishra** [comment]
@petebue , Thanks for the pointer. I did find that i forgot to list the IP Address (10.32.0.1) on the kube-apiserver cert. 

So, the DNS server comes up fine now. But when i do a nslookup it fails to serve the record

root@k8s-slave-1:~# nslookup kubernetes
Server:		10.0.2.3
Address:	10.0.2.3#53

** server can't find kubernetes: SERVFAIL

I am going to continue to debug this, but thanks for your help.

Closing out this issue because the pod has started successfully
